tests/unit/testsuite/Bolt/Boltpay/Helper/CouponTest.php

```diff
@@ -0,0 +1,572 @@
+
+    public function setUp()
+    {
+        $this->app = Mage::app('default');
+        $this->app->getStore()->resetConfig();
+        $this->couponHelper = new Bolt_Boltpay_CouponHelper();
+        $this->currentMock = new Bolt_Boltpay_Helper_Coupon();
+    }
+
+
+    protected function tearDown()
+    {
+        Mage::getSingleton('checkout/cart')->truncate()->save();
+    }
+
+    /**
+     * Generates data for testing purposes
+     */
+    public static function setUpBeforeClass()
+    {
+        try {
+            self::$ruleId = Bolt_Boltpay_CouponHelper::createDummyRule(self::$couponExistCode);
+            self::$invalidRuleId = Bolt_Boltpay_CouponHelper::createDummyRule(
+                self::$invalidCouponCode,
+                array(
+                    'from_date'         => (new \DateTime('now +1 day'))->format('Y-m-d'),
+                    'to_date'           => '2000-01-01',
+                    'uses_per_customer' => 1
+                ),
+                array(
+                    'usage_limit'        => 100,
+                    'times_used'         => 100,
+                    'usage_per_customer' => 1
+                )
+            );
+            self::$customerId = Bolt_Boltpay_CouponHelper::createDummyCustomer();
+            self::$quoteId = Bolt_Boltpay_CouponHelper::createDummyQuote(
+                array('customer_id' => self::$customerId, 'coupon_code', self::$invalidCouponCode)
+            );
+            self::$couponCodeId = Bolt_Boltpay_CouponHelper::getCouponIdByCode(self::$invalidCouponCode);
+            self::$productId = Bolt_Boltpay_ProductProvider::createDummyProduct('PHPUNIT_TEST_1');
+        } catch (\Exception $e) {
+            self::tearDownAfterClass();
+        }
+    }
+
+    /**
+     * Deletes dummy data
+     */
+    public static function tearDownAfterClass()
+    {
+        Bolt_Boltpay_CouponHelper::deleteDummyCustomer(self::$customerId);
+        Bolt_Boltpay_CouponHelper::deleteDummyRule(self::$ruleId);
+        Bolt_Boltpay_CouponHelper::deleteDummyRule(self::$invalidRuleId);
+        Bolt_Boltpay_CouponHelper::deleteDummyQuote(self::$quoteId);
+        Bolt_Boltpay_ProductProvider::deleteDummyProduct(self::$productId);
+    }
+
+    /**
+     * Sets up environment for discount test cases
+     *
+     * @param array $additionalData
+     *
+     * @throws Exception
+     */
+    private function setupEnvironment($additionalData = array())
+    {
+        $requestObject = $this->couponHelper->setUpRequest($additionalData);
+        $this->currentMock->setupVariables($requestObject);
+    }
+
+    /**
+     * Unit test for validating empty coupon code from request if it is passed
+     */
+    public function testPassValidateEmptyCoupon()
+    {
+        $check = true;
+        try {
+            $this->setupEnvironment(array('discount_code' => 'BOLT_CODE'));
+            $this->currentMock->validateEmptyCoupon();
+        } catch (\Exception $e) {
```

Comment:
Since you are expecting a very specific exception in the case of failures, `Bolt_Boltpay_BadInputException`, it is better to only catch that type of exception in your unit test.  In this case, by catching a generic Exception, you may inadvertently be allowing for subtle bugs to slip through the cracks.

Also consider if maybe `$passed` , `$hasPassed`, `$validated`, or `$isValid` is a better name for this variable.

I will only write this here, but this applies to all of the test that you set your `$check` in this way.